python/lsst/cbp/coordinateConverter.py

```diff
@@ -0,0 +1,856 @@
+        focalPlane1 = self._fieldAngleToFocalPlane.applyForward(Point2D(*pupilFieldAngle1))
+        focalPlane2 = self._fieldAngleToFocalPlane.applyForward(Point2D(*pupilFieldAngle2))
+        deltaFocalPlane = np.subtract(focalPlane2, focalPlane1)
+        return -math.atan2(deltaFocalPlane[1], deltaFocalPlane[0])*radians
+
+    def _getBeamCbpFieldAngle(self, beam):
+        """Return the field angle of the specified beam with respect to
+        the CBP pupil frame
+
+        Parameters
+        ----------
+        beam : `int`, `str` or None
+            Name or index of beam; if None then
+            ``self.maskInfo.defaultBeam``
+
+        Returns
+        -------
+        fieldAngle : a pair of floats, in radians
+        """
+        holePos = self.maskInfo.getHolePos(beam)
+        return tuple(math.atan(pos / self.config.cbpFocalLength) for pos in holePos)
+
+    def _computeBeamVectorInCtrPupilFrame(self, beamPosAtCtr, pupilFieldAngle):
+        """Compute the beam vector to the CBP in the centered pupil frame
+
+        beamPosAtCtr: position of beam on centered pupil (x,y mm)
+        pupilFieldAngle: incident angle of beam on pupil (x,y rad)
+        cbpDistance: distance between telescope and CBP centers (mm)
+        """
+        # compute beam position on "center pupil": a plane parallel to pupil at center of telescope
+        beamPosVec = coordUtils.pupilPositionToVector(beamPosAtCtr, self.config.telFlipX)
+        beamUnitVec = coordUtils.fieldAngleToVector(pupilFieldAngle, self.config.telFlipX)
+        abyz = beamPosVec[1]*beamUnitVec[1] + beamPosVec[2]*beamUnitVec[2]
+        beamPosMag = np.linalg.norm(beamPosAtCtr)
+        cbpDistance = self.config.cbpDistance
+        beamLength = -abyz + math.sqrt(math.fsum((cbpDistance**2, abyz**2, -beamPosMag**2)))
+        return beamLength*np.array(beamUnitVec)
+
+    def _computeCbpVectorInCtrPupilFrame(self, beamPosAtCtr, beamVectorInCtrPupil):
+        """Compute the vector from telescope to CBP in the telescope's
+        centered pupil frame
+
+        beamPosAtCtr: position of beam on centered pupil (x,y mm)
+        beamAngle: incident angle of beam on pupil (x,y rad)
+        """
+        beamPosVec = coordUtils.pupilPositionToVector(beamPosAtCtr, self.config.telFlipX)
+        return beamVectorInCtrPupil + beamPosVec
+
+    def _rotateFocalPlaneToPupil(self, focalPlane):
+        """Rotate a position or field angle in telescope focal plane
+        orientation to one in telescope pupil orientation
+        """
+        return coordUtils.rotate2d(pos=focalPlane, angle=-self._telRot)
+
+    def _rotatePupilToFocalPlane(self, pupil):
+        """Rotate a position or field angle in telescope pupil orientation
+        to one in telescope focal plane orientation
+        """
+        return coordUtils.rotate2d(pos=pupil, angle=self._telRot)
+
+    def getBeamInfo(self, beam, *, holePos=None):
```

Comment:
It means that if you want to provide a hole position (instead of accepting the default value) you must specify it by name. In my opinion that will make calling code much easier to understand, since specifying a hole position is a very unusual thing to do.

Reply:
Ah, right, that's interesting, and nice. I'm still learning the details of this operator, and honestly had also momentarily forgotten the fact that one can specify `kwargs` positionally as long as you don't start and stop (if that weren't true then this _wouldn't_ be doing anything, hence the comment). I like the fact that this can now be forced 👍